### PR TITLE
feat: remove mplex

### PR DESCRIPTION
**Pull request by hacdias**
Largely based on #10051. Requires #10068.

Boxo was updated with https://github.com/ipfs/boxo/pull/431, which updates `go-libp2p`, which removed support for mplex.

### CODE REVIEW

**Review by marten-seemann — APPROVED** [review]
Kill it with 🔥!

File: core/node/libp2p/smux.go [hunk 1]
```diff
@@ -3,52 +3,25 @@ package libp2p
...
-		return prioritizeOptions([]priorityOption{{
-			priority:        tptConfig.Multiplexers.Yamux,
-			defaultPriority: 100,
-			opt:             libp2p.Muxer(yamux.ID, yamux.DefaultTransport),
-		}, {
-			priority:        tptConfig.Multiplexers.Mplex,
-			defaultPriority: config.Disabled,
-			opt:             libp2p.Muxer(mplex.ID, mplex.DefaultTransport),
-		}}), nil
+		return nil, fmt.Errorf("configuring muxers with LIBP2P_MUX_PREFS is no longer supported")
```
Comment (lidel) [inline core/node/libp2p/smux.go:15]:
```suggestion
		return nil, fmt.Errorf("configuring muxers with LIBP2P_MUX_PREFS is no longer supported, use Swarm.Transports.Multiplexers")
```

File: core/node/libp2p/smux.go [hunk 2]
```diff
@@ -3,52 +3,25 @@ package libp2p
...
+	}
+	if tptConfig.Multiplexers.Mplex != 0 {
+		return nil, fmt.Errorf("Swarm.Transports.Multiplexers.Mplex is no longer supported")
```
Comment (lidel) [inline core/node/libp2p/smux.go:18]: If someone explicitly enabled it at some point and forgot, let's give people a pointer how to fix and why:
```suggestion
		return nil, fmt.Errorf("Swarm.Transports.Multiplexers.Mplex is no longer supported, remove it from your config, see https://github.com/libp2p/specs/issues/553")
```

File: core/node/libp2p/smux.go [hunk 3]
```diff
@@ -3,52 +3,25 @@ package libp2p
...
 	}
+	if tptConfig.Multiplexers.Yamux < 0 {
+		return nil, fmt.Errorf("Swarm.Transports.Multiplexers.Yamux is disabled even tho it is the only multiplexer available")
```
Comment (lidel) [inline core/node/libp2p/smux.go:21]:
```suggestion
		return nil, fmt.Errorf("running libp2p with Swarm.Transports.Multiplexers.Yamux disabled is not supported")
```

File: docs/config.md [hunk 4]
```diff
@@ -2118,19 +2118,7 @@ Type: `priority`
 
 **DEPRECATED**: See https://github.com/ipfs/kubo/issues/9958
```
Comment (lidel) [inline docs/config.md:2119]:
```suggestion
**REMOVED**: See https://github.com/ipfs/kubo/issues/9958
```

**Comment by Jorropo** [comment]
I forgot to update after our yesterday meeting, We will do that in v0.24 for 0.23 we are either gonna use Ethereum's V2 mplex or copy the old impl temporarily in an internal Kubo package.

**Review by lidel — APPROVED** [review]
Thank you @hacdias, lgtm, small documentation tweaks suggested inline, but ok to merge.

**Comment by hacdias** [comment]
@Jorropo please take that on then. I will keep this PR open until you have that ready, since I need it to test other PRs.

**Comment by hacdias** [comment]
Closing this now. It can be revived when we actually fully remove Mplex the next release.

**Comment by BigLep** [comment]
Full removal of mplex is in https://github.com/ipfs/kubo/issues/10069

**Pull request closed without merging**